CRM/Member/Form/MembershipStatus.php

```diff
@@ -51,15 +52,70 @@ public function getDefaultContext() {
     return 'create';
   }
 
+  /**
+   * Fields for the entity to be assigned to the template.
+   *
+   * Fields may have keys
+   *  - name (required to show in tpl from the array)
+   *  - description (optional, will appear below the field)
+   *  - not-auto-addable - this class will not attempt to add the field using addField.
+   *    (this will be automatically set if the field does not have html in it's metadata
+   *    or is not a core field on the form's entity).
+   *  - help (optional) add help to the field - e.g ['id' => 'id-source', 'file' => 'CRM/Contact/Form/Contact']]
+   *  - template - use a field specific template to render this field
+   * @var array
+   */
+  protected $entityFields = [];
+
+  /**
+   * Set entity fields to be assigned to the form.
+   */
+  protected function setEntityFields() {
+    $this->entityFields = [
+      'label' => [
+        'name' => 'label',
+        'description' => ts("Display name for this Membership status (e.g. New, Current, Grace, Expired...)."),
+        'required' => TRUE,
+      ],
+      'is_admin' => [
+        'name' => 'is_admin',
+        'description' => ts("Check this box if this status is for use by administrative staff only. If checked, this status is never automatically assigned by CiviMember. It is assigned to a contact's Membership by checking the <strong>Status Override</strong> flag when adding or editing the Membership record. Start and End Event settings are ignored for Administrator statuses. EXAMPLE: This setting can be useful for special case statuses like 'Non-expiring', 'Barred' or 'Expelled', etc."),
+      ],
+    ];
+  }
+
+  /**
+   * Set the delete message.
+   *
+   * We do this from the constructor in order to do a translation.
+   */
+  public function setDeleteMessage() {
+    $this->deleteMessage = ts('WARNING: Deleting this option will result in the loss of all membership records of this status.') . ' ' . ts('This may mean the loss of a substantial amount of data, and the action cannot be undone.') . ' ' . ts('Do you want to continue?');
```

Comment:
@eileenmcnaughton I realise that this is just a copy of the original message, but it's not clear to me what will happen - does it really mean that actual memberships will be deleted?
In fact, looking in `CRM_Member_BAO_MembershipStatus::del()` it would seem that a) it's only referring to the membershipstatus, and b) it won't delete anything if there are memberships with this status.

I've just confirmed that to be the case.
Something like:

> "You will not be able to delete this membership status if there are existing memberships with this status.  You will need to check all your membership status rules afterwards to ensure that a valid status will always be available.  Please confirm that you wish to delete this status?"